### PR TITLE
#714 prevent throwing another exception if the request fails eg connection reset

### PR DESCRIPTION
**Pull request by davidvanlaatum**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.0.x`. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

#714 fix to prevent calling close on nil object

@cliffano @zlx

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/ruby-client/api_client.mustache [hunk 1]
```diff
@@ -238,7 +238,7 @@ module {{moduleName}}
         tempfile.write(chunk)
       end
       request.on_complete do |response|
-        tempfile.close
+        tempfile.close if tempfile
         @config.logger.info "Temp file written to #{tempfile.path}, please copy the file to a proper folder "\
```
Comment (davidvanlaatum) [inline modules/openapi-generator/src/main/resources/ruby-client/api_client.mustache]: Side note shouldn't there be a way to turn this off?

**Comment by davidvanlaatum** [comment]
disabling logging of info doesn't feel like the right way to do it. Feels more like a dev hint that should go  away once its been seen

**Comment by davidvanlaatum** [comment]
for the build failure should I remove the 123_ that got added in a bunch of places?

**Comment by wing328** [comment]
> for the build failure should I remove the 123_ that got added in a bunch of places? I assume thats a different problem

No worry. Let me fix that.

**Comment by wing328** [comment]
UPDATE: I've filed https://github.com/OpenAPITools/openapi-generator/pull/719 to fix the Travis CI failure.

**Comment by wing328** [comment]
@davidvanlaatum The fix has been merged into master. Thanks for your contribution.

**Pull request merged**